plugins/net_plugin/include/eos/net_plugin/protocol.hpp

```diff
@@ -7,23 +7,26 @@ namespace eos {
    using namespace chain;
    using namespace fc;
 
-  struct handshake_message {
-      int16_t         network_version = 0;
-      chain_id_type   chain_id; ///< used to identify chain
-      fc::sha256      node_id; ///< used to identify peers and prevent self-connect
-      string          p2p_address;
-      uint32_t        last_irreversible_block_num = 0;
-      block_id_type   last_irreversible_block_id;
-      uint32_t        head_num = 0;
-      block_id_type   head_id;
-      string          os;
-      string          agent;
-   };
-
+   static_assert(sizeof(std::chrono::system_clock::duration::rep) >= 8, "system_clock is expected to be at least 64 bits");
    typedef std::chrono::system_clock::duration::rep tstamp;
-   typedef int32_t                                  tdist;
 
-   static_assert(sizeof(std::chrono::system_clock::duration::rep) >= 8, "system_clock is expected to be at least 64 bits");
+   struct handshake_message {
+      int16_t                    network_version = 0;
+      chain_id_type              chain_id; ///< used to identify chain
+      fc::sha256                 node_id; ///< used to identify peers and prevent self-connect
```

Comment:
peers should be identified by their authenticated public key even if randomly generated.

Reply:
Sorry, you'll need to be a little less terse than that.  I don't know what that means.

Reply:
Per Dan, node_id is to be removed at a later date.  I left it in in order to ease the merge with Phil's ongoing sync work.  Authentication makes it redundant.